src/protocol/graphics.ts

```diff
@@ -193,7 +218,11 @@ class VideoPlayer {
 export const Video = new VideoPlayer();
 
 let onRefreshGraphics: (canvas: Canvas) => void;
-let paintPointsWaiter: Promise<findPaintsResult>;
+let paintPointsWaiter: Deferred<void> = defer();
+let hasAllPaintPoints = false;
+paintPointsWaiter.promise.then(() => {
```

Comment:
It doesn't look like this ever resolves?

Reply:
Good point. This is a combination of a couple of different implementations that I tried 😓